l2-contracts/src/L2RewardManager.sol

```diff
@@ -99,7 +102,17 @@ contract L2RewardManager is
             recipient = recipient == address(0) ? claimOrders[i].account : recipient;
 
             // if the custom claimer is set, then transfer the tokens to the set claimer
-            XPUFETH.transfer(recipient, amountToTransfer);
+
+            //TODO: we have XPUFTH in this contract from previous transfer, so how do we handle this?
```

Comment:
Yeah, as discussed.
Give out the xPufETH, and when we are out of it, start giving away pufETH(oft)